example_client/rest_stateful_client.py

```diff
@@ -0,0 +1,449 @@
+            # result includes a dictionary with all model outputs
+            result = requests.post("{}:{}/v1/models/{}{}:predict".format(args['rest_url'], args['rest_port'], args['model_name'], version), data=data_json, cert=certs, verify=verify_server)
+            end_time = datetime.datetime.now()
+
+            try:
+                result_dict = json.loads(result.text)
+            except ValueError:
+                print("The server response is not json format: {}",format(result.text))
+                exit(1)
+            if "error" in result_dict:
+                print('Server returned error: {}'.format(result_dict))
+                exit(1)
+
+            if not validate_output(result_dict, output_names):
+                print(
+                    "ERROR: Model result validation error. Adding end sequence inference request for the model and exiting.")
+                exit(1)
+
+            # Unique sequence_id provided by OVMS
+            if get_sequence_id:
+                sequence_id = np.uint64(result_dict['sequence_id'])
+                get_sequence_id = False
+
+            duration = (end_time - start_time).total_seconds() * 1000
+            processing_times = np.append(
+                processing_times, np.array([int(duration)]))
+
+            # Compare results after we are pass initial context window results
+            if score_index >= 0:
+                # Loop over reference output results
+                avg_rms_error_sum = 0.0
+
+                for output_name in output_names:
+                    score_data = reference_scores[output_name][sequence_name][score_index]
+
+                    # Parse output
+                    results_array = make_ndarray(result_dict[output_name])
+
+                    # Calculate error
+                    avg_rms_error = calculate_utterance_error(
+                        score_data, results_array[0])
+                    avg_rms_error_sum += avg_rms_error
+
+                    # Statistics
+                    print_debug(
+                        'Output name: {} Rms error: {:.10f}\n'.format(
+                            output_name, avg_rms_error))
+
+                mean_avg_rms_error_sum += avg_rms_error_sum
+                # Statistics
+                print_debug(
+                    'Iteration {}; Average rms error: {:.10f} Processing time: {:.2f} ms; speed {:.2f} fps\n'.format(
+                        x, avg_rms_error_sum, round(
+                            np.average(duration), 2), round(
+                            1000 * sequence_size / np.average(duration), 2)))
+                # END output names loop
+
+            score_index += 1
+            # END utterance loop
+
+        seq_avg_rms_error_sum = mean_avg_rms_error_sum / (sequence_size)
+        print(
+            "\tSequence {} average rms error: {:.10f}\n".format(
+                sequence_id,
+                seq_avg_rms_error_sum))
+        sequence_id += 1
```

Comment:
Can we please get rid of this line? I think as we let user define sequence id in CLI argument we should stick to it the entire time.